ar_module2_8.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+var myDate = (function iffe() {
+  /* eslint-disable no-console */
+  /* eslint-disable no-unused-vars */
+  var privateDate = new Date(); // make this available to both get and set
+  var dd = privateDate.getDate();
+  var mm = privateDate.getMonth() + 1; //January is 0!
+  var yyyy = privateDate.getFullYear();
+
+  if (dd < 10) {
+    dd = '0' + dd;
+  }
+  if (mm < 10) {
+    mm = '0' + mm;
+  }
+
+  var checkDate = mm + dd + yyyy;
+  var formatDate = mm + '/' + dd + '/' + yyyy;
+
+  function setDate(newDate) {  // make a function for milli
+
+    if (newDate == undefined) {
+      var defaultDate = privateDate;
+      console.log(defaultDate);
+      // Done - myDate.setDate() if no argument given
+    }
+
+    else {  // only using newMilli in use
+      var milliDate = new Date();
+      milliDate.setDate(newDate);
+      console.log(milliDate);
+    }
+
+  }
+
+  function getDate(dateAsk) {
+    if (dateAsk == undefined) {
+      console.log('No Args');
+      var n = privateDate.getTime();
+      console.log(n);
+    }
+    else if (dateAsk == 'milliseconds') {
```

Comment:
The method should take an object as a parameter, and you want to check a property on that object called "format" to see what value it is set to. So inside your getDate implementation, your check will look something like this:

`if (dateAsk && dateAsk.format && dateAsk.format === 'milliseconds') {  
  // do something
} else if (dateAsk && dateAsk.format && dateAsk.format === 'formatted') {
  // do something
}`

Be sure to review the instructions to see the specifications for the method.